applications/GeoMechanicsApplication/CMakeLists.txt

```
@@ -0,0 +1,167 @@
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_constitutive/small_strain_umat_3D_interface_law.cpp
+
+    # Elements
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/U_Pw_base_element.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/U_Pw_small_strain_element.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/U_Pw_small_strain_interface_element.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/U_Pw_small_strain_link_interface_element.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/U_Pw_small_strain_FIC_element.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/small_strain_U_Pw_diff_order_element.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/drained_U_Pw_small_strain_element.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/undrained_U_Pw_small_strain_element.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/U_Pw_updated_lagrangian_element.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/updated_lagrangian_U_Pw_diff_order_element.cpp
+
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/geo_structural_base_element.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/geo_cable_element_3D2N.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/geo_cr_beam_element_2D2N.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/geo_cr_beam_element_3D2N.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/geo_cr_beam_element_linear_2D2N.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/geo_truss_element_3D2N.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/geo_truss_element_linear_3D2N.cpp
+
+    # Conditions
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/U_Pw_condition.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/U_Pw_force_condition.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/U_Pw_face_load_condition.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/U_Pw_normal_face_load_condition.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/U_Pw_normal_flux_condition.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/U_Pw_normal_flux_FIC_condition.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/U_Pw_face_load_interface_condition.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/U_Pw_normal_flux_interface_condition.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/general_U_Pw_diff_order_condition.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/line_load_2D_diff_order_condition.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/line_normal_load_2D_diff_order_condition.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/line_normal_fluid_flux_2D_diff_order_condition.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/surface_load_3D_diff_order_condition.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/surface_normal_load_3D_diff_order_condition.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/surface_normal_fluid_flux_3D_diff_order_condition.cpp
+
+    # Python application
+
+    # Strategies
+)
+
+if(${USE_PORO_MPI} MATCHES ON)
```

we have `USE_MPI` for that
```suggestion
if(${USE_MPI} MATCHES ON)
```